src/primitives/ray.jl

```diff
@@ -42,3 +42,13 @@ origin(r::Ray) = r.p
 The direction of the ray.
 """
 direction(r::Ray) = r.v
+
+function Base.in(p::Point, r::Ray)
+	w = norm(r.v)
+	d = evaluate(Euclidean(), p, Line(r(0), r(1)))
+	if d+w ≈ w
+		return sum((p - r.p) .* r.v) >= 0 # return true if point is in ray
```

Comment:
The only thing one has to take into account here, compared to `Base.in(p::Point, l::Line)` is to check whether the ray is hitting the point. `p - r.p` yields the vector pointing from the origin to the point (could be zero)
Multiplication with `r.v` yields positive numbers if it is in the direction otherwise negative